### PR TITLE
Add structured output for file edits

### PR DESCRIPTION
**Pull request by ryanhoangt**
## Description

This PR is to:
- Add structure to the output returned by `file_editor`, make it possible to parse and process in OH.
- Fix bug: missing exception in raise statement. 

## Related Issue

Progress towards https://github.com/All-Hands-AI/OpenHands/issues/4274

### CODE REVIEW

File: openhands_aci/editor/editor.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def __call__(
         insert_line: int | None = None,
         enable_linting: bool = False,
         **kwargs,
-    ) -> ToolResult | CLIResult:
+    ) -> ToolResult | CLIResult | FileModifyResult:
```
[the remark this answers is not in the document]
Comment (ryanhoangt) [reply openhands_aci/editor/editor.py]: Yeah this is a bit redundant... I used `CLIResult` only after collapsing `FileModifyResult` with it.

File: openhands_aci/editor/results.py [hunk 2]
```diff
@@ -14,11 +14,30 @@ class ToolResult:
...
+        return result
+
+
+@dataclass
 class CLIResult(ToolResult):
     """A ToolResult that can be rendered as a CLI output."""
 
 
+@dataclass
+class FileModifyResult(CLIResult):
```
[the remark this answers is not in the document]
Comment (ryanhoangt) [reply openhands_aci/editor/results.py]: Done -- can you have a look again?

File: openhands_aci/editor/results.py [hunk 3]
```diff
@@ -14,10 +14,25 @@ class ToolResult:
...
+        if extra_field:
+            result.update(extra_field)
+        return result
+
+
+@dataclass
 class CLIResult(ToolResult):
     """A ToolResult that can be rendered as a CLI output."""
 
+    # Opional fields for file editing commands
```
Comment (enyst) [inline openhands_aci/editor/results.py:30]:
```suggestion
    # Optional fields for file editing commands
```

File: openhands_aci/editor/results.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (BradKML) [reply openhands_aci/editor/results.py]: Can't wait to see the updates, cus the roadblock is a bit annoying

**Review by xingyaoww — APPROVED** [review]
LGTM! Just to confirm, you only move the tests to a separate file, but didn't modify most of them, right?

**Comment by ryanhoangt** [comment]
Yep!

**Comment by xingyaoww** [comment]
Sounds good! Seems good to merge :)

**Pull request merged**